imagej/imagej.py

```diff
@@ -344,7 +344,7 @@ def run_script(self, language, script, args=None):
             exts = script_lang.getExtensions()
             if exts.isEmpty():
                 raise ValueError("Script language '" + script_lang.getLanguageName() + "' has no extensions")
-            ext = exts.get(0)
+            ext = str(exts.get(0))
             try:
                 if args is None:
                     return self._ij.script().run("script." + ext, script, True).get()
```

Comment:
Yeah this is a fairly common problem.  Both Java and Python failed to define an interface for a String like object vs a concrete type.   Because many objects implement "str()" that does not qualify as string like for the purpose of protocols.  Java added a CharacterSequence interface, but it is so poorly used verses the concrete type.  As all of the methods in Python and Java assume they are getting the concrete type they lack any abstraction to work with other types well.  I can overload the wrapper type to play well with strings but I cannot make existing methods of string play well with JString.

Thus this type of casting is often needed.   The old method where JString would return as a Python string type was very performance limited and prevented call chaining of Java calls.  

```

PyObject *
PyUnicode_Concat(PyObject *left, PyObject *right)
{
    PyObject *result;
    Py_UCS4 maxchar, maxchar2;
    Py_ssize_t left_len, right_len, new_len;

    if (ensure_unicode(left) < 0)
        return NULL;

    if (!PyUnicode_Check(right)) {   // <== Does not check if type wants to be treated as a String
        PyErr_Format(PyExc_TypeError,
                     "can only concatenate str (not \"%.200s\") to str",
                     Py_TYPE(right)->tp_name);
        return NULL;
    }
    if (PyUnicode_READY(right) < 0)
        return NULL;
``` 

I tried to make JString an "unready" uncode object but there are not enough hooks in place to make this an option.